scripts/menu/server/sv_menu.lua

```diff
@@ -0,0 +1,405 @@
+  local src = source
+  local allow = PlayerHasTxPermission(src, 'players.teleport')
+  if allow then
+    TriggerClientEvent('txAdmin:menu:tpToWaypoint', src)
+    Wait(250)
+    local coords = GetEntityCoords(GetPlayerPed(src))
+    TriggerEvent("txaLogger:menuEvent", src, "teleportWaypoint", true,
+      { x = coords[1], y = coords[2], z = coords[3] })
+  else
+    TriggerEvent("txaLogger:menuEvent", src, "teleportWaypoint", false)
+  end
+end)
+
+RegisterServerEvent('txAdmin:menu:sendAnnouncement', function(message)
+  local src = source
+  local allow = PlayerHasTxPermission(src, 'players.message')
+  TriggerEvent("txaLogger:menuEvent", src, "announcement", allow, message)
+  if allow then
+    TriggerClientEvent('txAdmin:receiveAnnounce', -1, message)
+  end
+end)
+
+RegisterServerEvent('txAdmin:menu:fixVehicle', function()
+  local src = source
+  local allow = PlayerHasTxPermission(src, 'menu.vehicle')
+  TriggerEvent("txaLogger:menuEvent", src, "vehicleRepair", allow)
+  if allow then
+    TriggerClientEvent('txAdmin:menu:fixVehicle', src)
+  end
+end)
+
+local CREATE_AUTOMOBILE = GetHashKey("CREATE_AUTOMOBILE")
+RegisterServerEvent('txAdmin:menu:spawnVehicle', function(model)
+  local src = source
+  if type(model) ~= 'string' then
+    error()
+  end
+  
+  local allow = PlayerHasTxPermission(src, 'menu.vehicle')
+  TriggerEvent("txaLogger:menuEvent", src, "spawnVehicle", allow, model)
+  if allow then
+    local ped = GetPlayerPed(src)
+    local veh = Citizen.InvokeNative(CREATE_AUTOMOBILE, GetHashKey(model), GetEntityCoords(ped));
+    TriggerClientEvent('txAdmin:menu:spawnVehicle', src, NetworkGetNetworkIdFromEntity(veh))
+  end
+end)
+
+local function getPlayersLicense(src)
+  for _, v in ipairs(GetPlayerIdentifiers(src)) do
+    if string.sub(v, 1, string.len("license:")) == "license:" then
+      return v:sub(string.len("license:") + 1)
+    end
+  end
+end
+
+--[[ Emit player list to clients ]]
+CreateThread(function()
+  while true do
+    local found = {}
+    
+    local players = GetPlayers()
+    for _, serverID in ipairs(players) do
+      local ped = GetPlayerPed(serverID)
+      local veh = GetVehiclePedIsIn(ped)
+      if veh and veh > 0 then
+        veh = NetworkGetNetworkIdFromEntity(veh)
+      else
+        veh = nil
+      end
+  
+      local health = math.ceil(((GetEntityHealth(ped) - 100) / 100) * 100)
+      found[#found + 1] = {
+        id = serverID,
+        health = health,
+        veh = veh,
+        pos = GetEntityCoords(ped),
+        username = GetPlayerName(serverID),
+        license = getPlayersLicense(serverID)
+      }
+      
+      -- Lets wait a tick so we don't have hitch issues
+      Wait(0)
+    end
+  
+    -- get the list of all players to send to
+    debugPrint("^4Sending ^3" .. #found .. "^4 users details to ^3" .. #adminPermissions .. "^4 admins^0")
+    for id, _ in pairs(adminPermissions) do
+      TriggerClientEvent('txAdmin:menu:setPlayerState', id, found)
```

Comment:
That's a _lot_ of data for a reliable event, extremely optimistic towards admins having good network connectivity that won't drop, say, 700 players x ~80 bytes per player every 5 seconds without requiring resends.

Latent events would help a bit, along with a lowered rate and _maybe_ eventually keeping some sort of delta (not resending names, e.g.) as events do tend to be _somewhat_ ordered and mainly eventually-consistent.